test/new_relic/agent/instrumentation/rails/active_job_subscriber.rb

```diff
@@ -0,0 +1,201 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../../../../test_helper'
+require 'new_relic/agent/instrumentation/active_job_subscriber'
+
+module NewRelic::Agent::Instrumentation
+  class RetryMe < StandardError; end
+  class DiscardMe < StandardError; end
+
+  class TestJob < ActiveJob::Base
+    retry_on RetryMe
+
+    discard_on DiscardMe
+
+    def perform(error = nil)
+      raise error if error
+
+      rand(1138)
+    end
+  end
+
+  class ActiveJobSubscriberTest < Minitest::Test
+    NAME = 'perform.active_job'
+    ID = 71741
+    SUBSCRIBER = NewRelic::Agent::Instrumentation::ActiveJobSubscriber.new
+
+    def test_start
+      in_transaction do |txn|
+        time = Time.now.to_f
+        SUBSCRIBER.start(NAME, ID, {job: TestJob.new})
+        segment = txn.segments.last
+
+        assert_in_delta time, segment.start_time
+        assert_equal 'Ruby/ActiveJob/default/perform', segment.name
+      end
+    end
+
+    def test_start_when_not_traced
+      SUBSCRIBER.state.stub :is_execution_traced?, false do
+        in_transaction do |txn|
+          SUBSCRIBER.start(NAME, ID, {})
+
+          assert_empty txn.segments
+        end
+      end
+    end
+
+    def test_start_with_exception_raised
+      logger = MiniTest::Mock.new
+
+      NewRelic::Agent.stub :logger, logger do
+        logger.expect :error, nil, [/Error during .* callback/]
+        logger.expect :log_exception, nil, [:error, ArgumentError]
+
+        in_transaction do |txn|
+          SUBSCRIBER.stub :start_segment, -> { raise 'kaboom' } do
+            SUBSCRIBER.start(NAME, ID, {})
+          end
+
+          assert_equal 1, txn.segments.size
+        end
+      end
+      logger.verify
+    end
```

Comment:
I think this one may also be a dupe of the notifications subscriber tests.

Reply:
Addressed with [6e2aaf5](https://github.com/newrelic/newrelic-ruby-agent/pull/1761/commits/6e2aaf5f00b3447670f68fe508a2d7c52bd6263d)